packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * WordPress dependencies
+ */
+import { select, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { Settings, Page } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import type { Action, ActionModal } from '@wordpress/dataviews';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from './utils';
+import type { CoreDataError, PostWithPermissions } from '../types';
+
+const PAGE_POST_TYPE = 'page';
+
+const SetAsHomepageModal: ActionModal< PostWithPermissions >[ 'RenderModal' ] =
+	( { items, closeModal, onActionPerformed } ) => {
+		const [ item ] = items;
+		const pageTitle = getItemTitle( item );
+		const siteSettings = select( coreStore ).getEntityRecord< Settings >(
+			'root',
+			'site'
+		);
+		const pageOnFront = siteSettings?.page_on_front || null;
+		const currentHomePage =
+			pageOnFront &&
+			select( coreStore ).getEntityRecord(
+				'postType',
+				'page',
+				pageOnFront
+			);
+		const pageForPosts = siteSettings?.page_for_posts;
+		const showOnFront = siteSettings?.show_on_front;
+		const currentHomePageTitle = getItemTitle( currentHomePage as Page );
+
+		const { editEntityRecord, saveEditedEntityRecord, saveEntityRecord } =
+			useDispatch( coreStore );
+		const { createSuccessNotice, createErrorNotice } =
+			useDispatch( noticesStore );
+
+		async function onSetPageAsHomepage( event: React.FormEvent ) {
+			event.preventDefault();
+
+			try {
+				// If selected page is set to draft, publish the page.
+				if ( item.status === 'draft' ) {
+					await saveEntityRecord( 'postType', 'page', {
+						...item,
+						status: 'publish',
+					} );
+				}
+
+				// Save new home page settings.
+				await editEntityRecord( 'root', 'site', undefined, {
```

Comment:
Why not use `saveEntityRecord` directly?

Reply:
I'm not sure! It looks like `saveEntityRecord` works well, so I've swapped this out here: https://github.com/WordPress/gutenberg/pull/65426/commits/0ca96b02da76a52d412a8095b7a3112069a2e3d0.